src/ape/api/providers.py

```diff
@@ -1046,7 +1047,12 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
     def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         try:
-            txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            account = self.account_manager[txn.sender]
+            if isinstance(account, ImpersonatedAccount) and account.is_contract:
```

Comment:
If it is helper, we did this already in `ape-hardhat`: https://github.com/ApeWorX/ape-hardhat/blob/main/ape_hardhat/provider.py#L398-L415

Notice something that is missing the exception handling for revert-detection.

Perhaps we could remove the `ape-hardhat` custom implementation after this.

Reply:
Yeah, seems like a better way to handle it

Reply:
@unparalleled-js there is no problem regarding revert-detection here, if you see if-else part is wrapped with try except and in except part VmError is already handled. I am only removing is_contract part to apply the ape-hardhat changes as logically if sender is self.unlocked_accounts that means the sender is ImpersonatedAccount

Reply:
Also in ape-hardhat, this send_transaction only used for unlocked_accounts and for others super call will make it work with send_raw_transaction. which is already here now. To me, they are equal.